command/results_gotest.go

```diff
@@ -121,33 +121,29 @@ func sendTestLogsAndResults(ctx context.Context, comm client.Communicator, logge
 	return nil
 }
 
-// AllOutputFiles creates a list of all test output files that will be parsed, by expanding
-// all of the file patterns specified to the command.
-func (c *goTestResults) allOutputFiles() ([]string, error) {
+// globFiles returns a unique set of files that match the given glob patterns.
+func globFiles(patterns ...string) ([]string, error) {
+	var matchedFiles []string
 
-	outputFiles := []string{}
-
-	// walk through all specified file patterns
-	for _, pattern := range c.Files {
+	for _, pattern := range patterns {
 		matches, err := filepath.Glob(pattern)
 		if err != nil {
-			return nil, errors.Wrap(err, "error expanding file patterns")
+			return nil, errors.Wrapf(err, "globbing file pattern '%s'", pattern)
 		}
-		outputFiles = append(outputFiles, matches...)
+		matchedFiles = append(matchedFiles, matches...)
 	}
 
-	// uniquify the list
-	asSet := map[string]bool{}
-	for _, file := range outputFiles {
-		asSet[file] = true
+	// Uniquify the matches
+	asSet := map[string]struct{}{}
+	for _, match := range matchedFiles {
+		asSet[match] = struct{}{}
 	}
-	outputFiles = []string{}
-	for file := range asSet {
-		outputFiles = append(outputFiles, file)
+	matchedFiles = nil
```

Comment:
I know this  is a typed variable so it's technically okay to set it to nil before appending, but I think it's bad style: []string{} tells you more.

Reply:
The type is already clear from the declaration, but done.